cairo/tests/utils/args_gen.py

```diff
@@ -95,12 +96,44 @@
     Transaction,
 )
 from ethereum.cancun.trie import BranchNode, ExtensionNode, InternalNode, LeafNode, Node
+from ethereum.cancun.vm.exceptions import StackOverflowError, StackUnderflowError
 from ethereum.cancun.vm.gas import MessageCallGas
 from ethereum.crypto.hash import Hash32
 from ethereum.exceptions import EthereumException
 from ethereum.rlp import Extended, Simple
 from tests.utils.helpers import flatten
 
+"""Extensions over types defined in EELS.
+
+This module provides extensions that allow generating Cairo arguments when the Cairo memory
+layout of a type cannot be expressed using a generic Python type.
+
+For example, the Stack type is a List[U256] in Python, but requires implementing a mutable
+data structure in Cairo. This is done using a Dict[Uint, U256] to store elements along with
+a length field tracking the number of elements in the stack. Since List already has a generic
+argument generation strategy, we need a custom strategy for Stack.
+"""
+
+
+@dataclass
+class Stack(List[U256]):
+    """
+    An extension of the List[U256] type that allows for the generation of a Cairo 'Stack' argument.
+    """
+
+    def __init__(self, items: List[U256] = []):
+        super().__init__()
+        self.extend(items)
+
+    def gen_arg(self, dict_manager: DictManager, segments: MemorySegmentManager):
```

Comment:
I'd rather just implement here the transformation of `Stack` into a default type (here a dict); call this `to_base_type()` and possibly add a mapping like `_type_aliases` (called e.g. `_type_to_base_types`) and hook for this in `gen_arg` to call `to_base_type`

Reply:
you can use a `from typing import Protocol` to define a type that can be used to tell that these types need to implement this `to_base_type` (like a trait)